src/components/common/text/CryptoText.js

```diff
@@ -0,0 +1,50 @@
+// @flow
+
+import { type EdgeCurrencyWallet } from 'edge-core-js'
+
+import { useCryptoText } from '../../../hooks/useCryptoText'
+import { useSelector } from '../../../types/reactRedux'
+import { getAllTokens } from '../../../util/CurrencyInfoHelpers'
+import { fixFiatCurrencyCode, getDenomFromIsoCode, zeroString } from '../../../util/utils'
+
+type Props = {
+  wallet: EdgeCurrencyWallet,
+  tokenId?: string,
+  nativeAmount: string
+}
+
+/**
+ * Returns a cleaned crypto amount string. If no tokenId is given, use the
+ * wallet's native currency.
+ **/
+export const CryptoText = ({ wallet, tokenId, nativeAmount }: Props) => {
+  const exchangeRates = useSelector(state => state.exchangeRates)
```

Comment:
It's a bad idea to select the entire 'exchangeRates' since that would trigger a render to ALL the TickerTexts every time ANY of them updates. 

Instead, you should be more precise in your selection.

You should do something like this - 

```js

const [fiatCurrencyCode, setFiatCurrencyCode] = useState(edgeWallet.fiatCurrencyCode)
  const isoFiatCurrencyCode = fixFiatCurrencyCode(fiatCurrencyCode)
  const rateKey = `${currencyCode}_${isoFiatCurrencyCode}`
  const yesterdayUsdRateKey = `${currencyCode}_iso:USD_${getYesterdayDateRoundDownHour()}`

  const rawExchangeRate = useSelector(state => state.exchangeRates[rateKey])
  const usdExchangeRate = useSelector(state => state.exchangeRates[`iso:USD_${isoFiatCurrencyCode}`])
  const yesterdayUsdExchangeRate = useSelector(state => state.exchangeRates[yesterdayUsdRateKey])
  
  ```
  
  
  
  Look at this file for reference: https://github.com/EdgeApp/edge-react-gui/blob/cfa44ebcc86e4316be7123357fdecdc8243cd571/src/components/themed/WalletListCurrencyRow.js